### ISSUE TITLE
Bracket on new line indentation  failure for switch expression that assigned a variable

### ISSUE THREAD

**Issue by AntoineLange** [checkstyle/checkstyle#11155]
```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="LeftCurly">
          <property name="option" value="nl"/>
      </module>
      <module name="Indentation"/>
  </module>
</module>

/var/tmp $ cat Test.java
public class Test
{
    String s1 = "a1";
    String s2 = switch (s1)
    {
        case "a1" -> "a2";
        case "b1" -> "b2";
        default -> "error";
    };
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-9.2.1-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:5:5: '{' has incorrect indentation level 4, expected level should be 8. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```

---

We use bracket on new line in our coding standards (like apache https://commons.apache.org/proper/commons-net/code-standards.html). That's why I use "leftCurly" option "nl" in my checkstyle configuration.
We just switch from Java 11 to Java 17 and try to use the new switch expression from Java 14. It went well except when we tried to assign the result of the switch to a new variable in the same statement.
If I assigned an empty variable one line before, I don't get an error, therefore this works:
```bash
public class Test
{
    String s1 = "a1";
    String s2;
    s2 = switch (s1)
    {
        case "a1" -> "a2";
        case "b1" -> "b2";
        default -> "error";
    };
}
```
---

**Comment by pree-T** [comment]
May I work on it? Can someone help me to fix it as I am new to open source contribution?

**Comment by romani** [comment]
Indentation is most complicated Check in checkstyle, better to avoid fixing it if you are not experienced.

**Comment by romani** [comment]
Everyone welcome to send Pull-request

**Comment by mahfouz72** [comment]
@nrmancuso @rnveach  I can't reproduce?

```
PS D:\CS\test> javac src/Test2.java                                                                                                
PS D:\CS\test> cat src/Test2.java
public class Test2
{
    String s1 = "a1";
    String s2 =switch(s1)
    {
        case "a1" -> "a2";
        case "b1" -> "b2";
        default -> "error";
    };
}
PS D:\CS\test> cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
        <module name="LeftCurly">
            <property name="option" value="nl"/>
        </module>
        <module name="Indentation"/>

    </module>
</module>
PS D:\CS\test> java  -jar checkstyle-10.17.0-all.jar -c config.xml  src/Test2.java
Starting audit...
Audit done.
PS D:\CS\test> 
```

**Comment by nrmancuso** [comment]
Ok, looks like we fixed this in the scope of https://github.com/checkstyle/checkstyle/pull/14024, I am closing this issue. @AntoineLange thanks for your bug report, please let us know if you run into some other case that we missed.
